README.md

```diff
@@ -5,3 +5,104 @@
+
+### Build containers
+
+So that tests are identical whether run locally, against Triton, or on Shippable, it's a good idea for tests to run in a build container. An example Dockerfile for a build container can be found in this repo: [Dockerfile.example](https://github.com/autopilotpattern/testing/blob/master/Dockerfile.example). This Dockerfile assumes that your application has test code found at `/tests/tests.py` as well as a `docker-compose.yml` file.
+
+### Testing against local Docker
+
+For running the build container locally and deploying the tests against a local Docker Engine (for example, Docker for Mac), you'll create your build container and then mount the `docker.sock` to it.
+
+```
+docker run -it --rm \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+	-e PATH=/root/venv/3.5/bin:/usr/bin:/usr/local/bin \
+	-e COMPOSE_HTTP_TIMEOUT=300 \
+	-w /src \
+    example/myBuildContainer \
+    python tests.py
+```
+
+This will start the build container, setup the Python3 environment, and then run your tests by sending Docker API commands along the `docker.sock` to the Docker engine on your laptop.
+
+### Testing on Triton
+
+For running the build container locally but deploying the tests against Triton, you'll create your build container and provide it with your credentials to Triton by mounting them to the test container and proving the appropriate environment variables. In the example below, the tests will be run on Joyent's Triton Cloud in the `us-sw-1` data center.
+
+```
+docker run -it --rm \
+	-v ~/.triton:/root/.triton \
+    -e DOCKER_TLS_VERIFY=1 \
+    -e DOCKER_CERT_PATH=/root/.triton/docker/username@us-sw-1_api_joyent_com \
+    -e DOCKER_HOST=tcp://us-sw-1.docker.joyent.com:2376 \
+	-e PATH=/root/venv/3.5/bin:/usr/bin:/usr/local/bin \
+	-e COMPOSE_HTTP_TIMEOUT=300 \
+	-w /src \
+    example/myBuildContainer \
+    python tests.py
+
+```
+
+### Testing on Shippable
+
+For our tests running on Shippable against Triton, there's a good deal more setup involved. You must have a Shippable account, and configure it with a Docker Hub integration and an ssh key integration. The key you provide Shippable must have permissions to deploy as a user on Triton. These integrations get added to the `shippable.yml` file:
+
+```yml
+integrations:
+  hub:
+    - integrationName: DockerHub
+      type: docker
+
+  key:
+    - integrationName: MyTritonTestingKey
+      type: ssh-key
+```
+
+We can't mount our credentials on our laptops to the build container on Shippable, so we need to use the ssh key integration to create a Triton profile using that key. While we could create a Triton profile and commit it to the GitHub repo safely (the private key isn't in our repo), it's better to dynamically generate it when we run the build. This makes it easy for anyone to set up their own Shippable test pipeline without hard-coded users.
```

Comment:
> the private key isn't in our repo

I think you mean that the key isn't part of the profile that's committed to the repo, but it took me some time to parse.